edi_oca/models/edi_exchange_record.py

```diff
@@ -237,7 +239,7 @@ def name_get(self):
         for rec in self:
             rec_name = rec.identifier
             if rec.res_id and rec.model:
-                rec_name = rec.record.display_name
+                rec_name = rec.record.sudo().display_name
```

Comment:
pls call sudo on self at the beginning of the loop (eg `for rec in self.sudo()`)